searx/engines/duckduckgo_extra.py

```diff
@@ -69,28 +73,68 @@ def request(query, params):
         args['p'] = safe_search  # "-1", "1"
 
     logger.debug("cookies: %s", params['cookies'])
+
+    path = 'i'
+    if ddg_category == 'videos':
+        path = 'v'
+    elif ddg_category == 'news':
+        path = 'news'
     args = urlencode(args)
-    params['url'] = 'https://duckduckgo.com/i.js?{args}'.format(args=args)
+    params['url'] = f'https://duckduckgo.com/{path}.js?{args}'
 
     return params
 
 
+def _image_result(result):
+    return {
+        'template': 'images.html',
+        'url': result['url'],
+        'title': result['title'],
+        'content': '',
+        'thumbnail_src': result['thumbnail'],
+        'img_src': result['image'],
+        'img_format': '%s x %s' % (result['width'], result['height']),
+        'source': result['source'],
+    }
+
+
+def _video_result(result):
+    return {
+        'template': 'videos.html',
+        'url': result['content'],
+        'title': result['title'],
+        'content': result['description'],
+        'thumbnail': result['images'].get('small') or result['images'].get('medium'),
+        'iframe_src': result['embed_url'],
+        'source': result['provider'],
+        'length': result['duration'],
+        'metadata': result.get('uploader'),
```

Comment:
https://docs.searxng.org/dev/engines/engine_overview.html#videos
We should consider adding ```metdata``` or ```extra``` field to all templates.
Which is a dict and it includes all kinds of metadata, even if they will not be used in the UI.

Also the ```ìframe_src``` should be added to docs.

Reply:
The macros `result_header`, `result_sub_header` and `result_sub_footer`:

https://github.com/searxng/searxng/blob/c3ab49cd903d27905d2da6f70699a55c9a74593e/searx/templates/simple/macros.html#L20-L48

are included in various templates / for instance in the `video.html` template:

https://github.com/searxng/searxng/blob/c3ab49cd903d27905d2da6f70699a55c9a74593e/searx/templates/simple/result_templates/videos.html#L3-L4

https://github.com/searxng/searxng/blob/c3ab49cd903d27905d2da6f70699a55c9a74593e/searx/templates/simple/result_templates/videos.html#L18-L18

> We should consider adding ```metdata``` or ```extra``` field to all templates.

we have a `metadata` field in the `result_sub_header` macro:

https://github.com/searxng/searxng/blob/c3ab49cd903d27905d2da6f70699a55c9a74593e/searx/templates/simple/macros.html#L34-L39